### DISCUSSION TITLE
Duda parte 1

### DISCUSSION THREAD

**Discussion by IgnacioHqz** [IIC2133-PUC/2024-2#173] [Tarea 3] [answered]
Hola, estoy algo confundido con el enunciado. En la parte 1 se menciona que cada paso que se da despierta ligeramente al puffle pero si se espera un poco antes de dar el siguiente paso el puffle vuelve a dormir, pero no entiendo el ejemplo 1. Se supone que al dar el primer paso la profundidad llega a 3, lo que sería menor a w (que es 4) y por lo tanto el puffle se despertaría, pero no entiendo por qué en el ejemplo se menciona que hay que esperar y que la profundidad requerida es 6. Cuál sería la explicación de que la profundidad sea 6? O será que antes de dar un paso primero hay que ver si es que al darlo quedo por debajo de w y en ese caso esperar el tiempo necesario para no caer por debajo de w? Yo asumí que primero se da el paso y despúes hay que esperar. Estoy algo confundido con ese ejemplo, si me pudieran aclarar lo agradecería mucho!

**Comment by JavierLarre** [comment, marked as answer]
Hola Ignacio! lo que debes hacer es **Al revés**: no puedes dar un paso que baje de la profundidad mínima, lo que queremos es **no** despertar al puffle. Te explico el ejemplo:
Si la profundida mínima es 4 y el paso que debemos tomar es de peso 2 entonces nuestra profundida actual **debe ser por lo menos 6 o más**, así el paso no despertará al puffle.  
PERO nuestra profundida inicial/máxima es 5, por lo tanto es imposible dar el paso. Por eso el ejemplo termina con -1.
El 6 solo aparece cuando queremos saber si podemos dar el paso.

Espero que te haya ayudado! cualquier duda la puedes consultar.
